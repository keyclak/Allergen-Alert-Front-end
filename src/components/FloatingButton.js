import React, { useEffect, useState } from 'react';
import { View, Text, Pressable, StyleSheet, ActivityIndicator, TextInput } from 'react-native';
import { Colors, StyleConstants, Styles } from '../style';

export default function FloatingButton({onPress, children}) {

    return (
        <Pressable style={[Styles.floatingButton, {bottom: StyleConstants.FloatingButtonSize / 2, right: StyleConstants.FloatingButtonSize / 2}]} onPress={onPress}>
            {children}
        </Pressable>
    )
}