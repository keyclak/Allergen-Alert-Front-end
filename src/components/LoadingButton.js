import React, { useEffect, useState } from 'react';
import { View, Text, Pressable, StyleSheet, ActivityIndicator, TextInput } from 'react-native';
import { Colors, Styles } from '../style';

export default function LoadingButton({children, isLoading, onPress, style}) {
    return (
        <Pressable style={[Styles.button, style]} onPress={onPress}>
            {
                isLoading
                    ? <ActivityIndicator color={Colors.AccentForeground} size="large"/>
                    : children
            }
        </Pressable>
    )
}