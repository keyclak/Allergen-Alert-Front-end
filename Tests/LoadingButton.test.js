import renderer from 'react-test-renderer';
import React from 'react';
import MockDate from 'mockdate'
import FloatingButton from '../src/components/FloatingButton';
import LoadingButton from '../src/components/LoadingButton';

const FRAME_TIME = 10

const advanceOneFrame = () => {
  const now = Date.now();
  MockDate.set(new Date(now + FRAME_TIME));
  jest.advanceTimersByTime(FRAME_TIME);
}

const timeTravel = (msToAdvance = FRAME_TIME) => {
  const numberOfFramesToRun = msToAdvance / FRAME_TIME;
  let framesElapsed = 0;

  while (framesElapsed < numberOfFramesToRun) {
    advanceOneFrame();
    framesElapsed++;
  }
}
global.requestAnimationFrame = cb => {
  setTimeout(cb, FRAME_TIME);
}

export const setupTimeTravel = () => {
    MockDate.set(0);
    jest.useFakeTimers();
}

jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');
jest.mock('@expo/vector-icons', () => ({
    MaterialIcons: 'Icon'
}));

beforeEach(setupTimeTravel);

describe("LoadingButtonTests", () => {
    test("Renders correctly", async () => {
        const tree = renderer.create(
            <LoadingButton children={jest.fn()} isLoading={jest.fn()} onPress={jest.fn()} style={jest.fn()} />
        ).toJSON(); 

        timeTravel(500);

        expect(tree).toMatchSnapshot();
    })
})