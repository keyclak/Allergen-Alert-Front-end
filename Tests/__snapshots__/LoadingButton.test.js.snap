// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingButtonTests Renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "backgroundColor": "#2AB7CA",
        "borderRadius": 15,
        "flexDirection": "row",
        "height": 50,
        "justifyContent": "center",
        "marginBottom": 15,
      },
      false,
      [MockFunction],
    ]
  }
>
  <ActivityIndicator
    animating={true}
    color="white"
    hidesWhenStopped={true}
    size="large"
    style={
      Object {
        "position": "absolute",
        "transform": Array [
          Object {
            "translateX": 200,
          },
        ],
      }
    }
  />
  <View
    style={
      Object {
        "position": "absolute",
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
      }
    }
  />
</View>
`;
